src/parser/Parser.ts

```diff
@@ -2375,6 +2369,54 @@ export class Parser {
         }
     }
 
+    /**
+     * References are found during the initial parse.
+     * However, sometimes plugins can modify the AST, requiring a full walk to re-compute all references.
+     * This does that walk.
+     */
+    findReferences() {
+        const references: References = {
```

Comment:
Should be a `validateReferences` which would run only if needed.

Reply:
I will change the signature of this method to `private`. Nobody should be calling this except the `Parser.references` getter, which has the logic to conditionally run this method.

Reply:
Makes sense now